grakn-core/src/main/java/ai/grakn/GraknTransaction.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn;
+
+/**
+ * <p>
+ *     A Class Desrcibing the behaviour of a transaction
+ * </p>
+ *
+ * <p>
+ *     This class is used to describe how a transaction on {@link GraknGraph} should behave.
+ *     When producing a graph using a {@link GraknSession} one of the following enums must be provided:
+ *         READ - A read only transaction. If you attempt to mutate the graph with such a transaction an exception will be thrown.
+ *         WRITE - A transaction which allows you to mutate the graph.
+ *         BATCH - A transaction which allows mutations to be performed more quickly but disables some consitency checks.
+ * </p>
+ *
+  @author fppt
+ */
+public enum GraknTransaction {
+    READ,  //Read only transaction where mutations to the graph are prohibited
```

Comment:
I think I called it that initially for the same reasons but then went for this shorter option because the length annoyed me when you say `session.open(GraknTransactionType.WRITE)`

I will change it back. I can get over my hatred of the length.